### ISSUE TITLE
TV Show/Movie Categories are empty

### ISSUE THREAD

**Issue by mo123** [Sandmann79/xbmc#180]
**Varstahl's edit for visibility**: it appears that some users (especially italians) are experiencing issues with the usage of the addon, in regard to categories being shown as empty. This might be due to a lack of DNS propagation and/or excessive DNS caching, or to a progressive user migration from the old servers to the new ones. A temporary workaround, by rollback to 0.5.7, is described in [this post below](https://github.com/Sandmann79/xbmc/issues/180#issuecomment-427599204).

---

Just reinstalled latest Kodi nightly from yesterday on Android and on Linux Mint.
Addon used was v0.5.9.
It shows tv shows being loaded at the top right, Amazon VOD... when you click TV Shows.
But when you go into a category like Animation, Kids, Action they are all still empty.
At the top right when you open a category it says 'Amazon VOD Loading Next page' 3 times without ever showing a show name.

Looks like there are a lot of errors in the log.

Log: 
https://pastebin.com/Y6tWZNVW

**Comment by Sandmann79** [comment]
After the last update, you have to close Kodi and remove the following catalog file:
`/storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/userdata/addon_data/plugin.video.amazon-test/PVCatalog[ID].pvcp`

**Comment by Allev8** [comment]
I got the same issue. I've delete the file but nothing change.

**Comment by Varstahl** [comment]
The logs provided don't seem to provide any information in regard to the addon.

@mo123 your installation seems off, there are a lot of errors like this:
```
WARNING: [Amazon VOD] Unable to decode date "October 24, 2005": language "None" not supported
```
That's not a major problem, but it should never happen. However since you're both experiencing the same issue, as reported by other users, this one time due to the heavy changes needed for the new UI, video data could actually be corrupted or in a state where it's no longer possible for it to work correctly. So on top of deleting the `PVCatalog[ID].pvcp` you also need to delete the `PVVideoData[ID].pvcp`.

This is not required for all the users, but if you're experiencing this kind of problem it's probably due to corrupt video data. Flushing both should fix the problem.

**Comment by Allev8** [comment]
I just did but still the same, all category are empty except continue watching and suggested for you. If i try to open action movie, for example, the pop up "Amazon VOD Loading Next page" shows up but nothing happen and it's empty, then if i try to reload the same category i see empty list right away without even try to reload it.

**Comment by Varstahl** [comment]
Did you close Kodi before removing the files?

**Comment by Allev8** [comment]
Yes and i tried in 2 others devices, these with coreelec, the first one is kodi running on nvidia shield.
Still the same for all three.

**Comment by Varstahl** [comment]
This is weird, we've tested under windows, debian and LibreElec. I'll need to look into it.

Can you do me a favour? If you can: close kodi, remove the catalog and video data, try to load one movie or tv series category until you get to an empty listing, and then post somewhere the log in its entirety. In the logs @mo123 provided there were no glaring issues, and in the log another user provided there were clear errors of video data corruption (pre-wipe). I can't seem to reproduce the issue in any way.

**Comment by Allev8** [comment]
Here the log https://pastebin.com/8LWm68tJ

**Comment by Allev8** [comment]
This is for coreelec https://pastebin.com/cSzdvbVj

**Comment by Varstahl** [comment]
`WARNING: CreateLoader - unsupported protocol(plugin) in plugin://plugin.video.amazon-test/[…]`

@Allev8 there seems to be some heavy conficts with something that tries to keep building invalid plugin references, mainly to resources like VIDEO_TS, which is a section on DVDs. Other than that I don't see any problem with Amazon VOD from the logs, except something very weird:

`https://raw.githubusercontent.com/Sandmann79/xbmc/master/plugin.video.amazon-test/resources/menu/.json`

@Sandmann79 shouldn't those be grabbed only with amazon.<tlds>?

---

Ok, one big problem in both of the logs:

```
ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
   - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
  Error Type: <type 'exceptions.IOError'>
  Error Contents: [Errno 2] No such file or directory: u'ATVPDKIKX0DER/ATVPDKIKX0DERsATVPDKIKX0DERtATVPDKIKX0DERoATVPDKIKX0DERrATVPDKIKX0DERaATVPDKIKX0DERgATVPDKIKX0DEReATVPDKIKX0DER/ATVPDKIKX0DER.ATVPDKIKX0DERkATVPDKIKX0DERoATVPDKIKX0DERdATVPDKIKX0DERiATVPDKIKX0DER/ATVPDKIKX0DERaATVPDKIKX0DERdATVPDKIKX0DERdATVPDKIKX0DERoATVPDKIKX0DERnATVPDKIKX0DERsATVPDKIKX0DER/ATVPDKIKX0DERpATVPDKIKX0DERlATVPDKIKX0DERuATVPDKIKX0DERgATVPDKIKX0DERiATVPDKIKX0DERnATVPDKIKX0DER.ATVPDKIKX0DERvATVPDKIKX0DERiATVPDKIKX0DERdATVPDKIKX0DEReATVPDKIKX0DERoATVPDKIKX0DER.ATVPDKIKX0DERaATVPDKIKX0DERmATVPDKIKX0DERaATVPDKIKX0DERzATVPDKIKX0DERoATVPDKIKX0DERnATVPDKIKX0DER-ATVPDKIKX0DERtATVPDKIKX0DEReATVPDKIKX0DERsATVPDKIKX0DERtATVPDKIKX0DER/ATVPDKIKX0DERrATVPDKIKX0DEReATVPDKIKX0DERsATVPDKIKX0DERoATVPDKIKX0DERuATVPDKIKX0DERrATVPDKIKX0DERcATVPDKIKX0DEReATVPDKIKX0DERsATVPDKIKX0DER/ATVPDKIKX0DERmATVPDKIKX0DEReATVPDKIKX0DERnATVPDKIKX0DERuATVPDKIKX0DER/ATVPDKIKX0DER.ATVPDKIKX0DERjATVPDKIKX0DERsATVPDKIKX0DERoATVPDKIKX0DERnATVPDKIKX0DER'
  Traceback (most recent call last):
    File "/storage/.kodi/addons/plugin.video.amazon-test/default.py", line 5, in <module>
      EntryPoint()
    File "/storage/.kodi/addons/plugin.video.amazon-test/resources/lib/startup.py", line 48, in EntryPoint
      loadUser('atvurl', cachedUsers=users), loadUser('pv', cachedUsers=users))
    File "/storage/.kodi/addons/plugin.video.amazon-test/resources/lib/common.py", line 124, in InitialiseProvider
      self._globals['amz'] = AmazonTLD(self, Settings())
    File "/storage/.kodi/addons/plugin.video.amazon-test/resources/lib/singleton.py", line 14, in __call__
      cls._instances[cls] = super(_Singleton, cls).__call__(*args, **kwargs)
    File "/storage/.kodi/addons/plugin.video.amazon-test/resources/lib/amazontld.py", line 23, in __init__
      self.loadCategories()
    File "/storage/.kodi/addons/plugin.video.amazon-test/resources/lib/amazontld.py", line 243, in loadCategories
      data = json.load(open(jsonfile))
  IOError: [Errno 2] No such file or directory: u'ATVPDKIKX0DER/ATVPDKIKX0DERsATVPDKIKX0DERtATVPDKIKX0DERoATVPDKIKX0DERrATVPDKIKX0DERaATVPDKIKX0DERgATVPDKIKX0DEReATVPDKIKX0DER/ATVPDKIKX0DER.ATVPDKIKX0DERkATVPDKIKX0DERoATVPDKIKX0DERdATVPDKIKX0DERiATVPDKIKX0DER/ATVPDKIKX0DERaATVPDKIKX0DERdATVPDKIKX0DERdATVPDKIKX0DERoATVPDKIKX0DERnATVPDKIKX0DERsATVPDKIKX0DER/ATVPDKIKX0DERpATVPDKIKX0DERlATVPDKIKX0DERuATVPDKIKX0DERgATVPDKIKX0DERiATVPDKIKX0DERnATVPDKIKX0DER.ATVPDKIKX0DERvATVPDKIKX0DERiATVPDKIKX0DERdATVPDKIKX0DEReATVPDKIKX0DERoATVPDKIKX0DER.ATVPDKIKX0DERaATVPDKIKX0DERmATVPDKIKX0DERaATVPDKIKX0DERzATVPDKIKX0DERoATVPDKIKX0DERnATVPDKIKX0DER-ATVPDKIKX0DERtATVPDKIKX0DEReATVPDKIKX0DERsATVPDKIKX0DERtATVPDKIKX0DER/ATVPDKIKX0DERrATVPDKIKX0DEReATVPDKIKX0DERsATVPDKIKX0DERoATVPDKIKX0DERuATVPDKIKX0DERrATVPDKIKX0DERcATVPDKIKX0DEReATVPDKIKX0DERsATVPDKIKX0DER/ATVPDKIKX0DERmATVPDKIKX0DEReATVPDKIKX0DERnATVPDKIKX0DERuATVPDKIKX0DER/ATVPDKIKX0DER.ATVPDKIKX0DERjATVPDKIKX0DERsATVPDKIKX0DERoATVPDKIKX0DERnATVPDKIKX0DER'
  -->End of Python script error report<--
```

There are traces of addon using both AmazonTLD and PrimeVideo. Something went terribly wrong, this is going to be deeply problematic. @Allev8, @mo123 what location are you guys in? Using PrimeVideo.com or amazon.de/co.uk/jp/com?

**Comment by Allev8** [comment]
I'm in Italy.

**Comment by Varstahl** [comment]
@Allev8 did you also use Amazon VOD 0.5.8, when the login was corrupted? If so, when you now tested it on other devices, did they previously have Amazon VOD or you just installed them freshly with 0.5.9?

**Comment by Allev8** [comment]
Yes Amazon VOD 0.5.8 was installed in all three devices and has always worked 'till 0.5.9.

**Comment by Varstahl** [comment]
Okay, the login and auto-detection is broken. So, it might be that the login data is corrupted, and that's where these problems are stemming out of. Do me a solid: if you're using multi-user with Amazon VOD, delete all the accounts and readd the main one, or just redo the login process if you're on a single account. Either way should restore the proper login data, unless Amazon changed that as well.

**Comment by Varstahl** [comment]
I just finished working and took a look at it: @Sandmann79, bug confirmed, the login process is broken. I don't know if I'll manage to debug/test it tonight, but there is something wrong with the login process, specifically things start getting weird after the territorial detection.

**Comment by Varstahl** [comment]
@Allev8 if you log in, quit kodi and restart kodi, then enter the addon, quit and then reenter (I know, it's convoluted) does it start working properly for you? I just did a test erasing everything, and although it's kind of broken somewhere in the beginning after the login, it works properly.

**Comment by i-am-bigmike** [comment]
I have the same problem, the lists are empty. (Everything worked fine until 3 October)

Not sure if it's helpful: I still see the movie titles in "Continue watching" and these play correctly. Even in "Suggested movies" I find a lot of titles, and I'm able to watch them too.

**Comment by Varstahl** [comment]
There seems to be a whole lot going on and I'm not understanding the half of it. I somehow believe though that if you check back in, everything should probably be working correctly now. Close Kodi, try to delete both cache file and try again. Seems stupid, but… (*am I going crazy?*)

**Comment by Allev8** [comment]
> 
> 
> @Allev8 if you log in, quit kodi and restart kodi, then enter the addon, quit and then reenter (I know, it's convoluted) does it start working properly for you? I just did a test erasing everything, and although it's kind of broken somewhere in the beginning after the login, it works properly.

Ok, I did a fresh install again then logged in directly in the settings, closed and restarted kodi then entered in the addon, exited and reentered and no, i still got empty list. I've even try to delete the 2 files after that but it still the same.

**Comment by Varstahl** [comment]
I'm starting to believe it was a fluke with the Amazon's API endpoints. I've had the same exact problem once, after that I've kept resetting everything (user, password, useragent, cache, videodata) and has since been working properly ever since. I don't know what happened but I can't reproduce the problem anymore.

If anyone is still having problems, can you please completely erase the content of `userdata/addon_data/plugin.video.amazon-test` (warning: this also erases settings) and test again?

I've erase all the data 10 times by now, and I only had problems the first time, I can't figure out what is going on for the life of me.

**Comment by Varstahl** [comment]
> Ok, I did a fresh install again then logged in directly in the settings, closed and restarted kodi then entered in the addon, exited and reentered and no, i still got empty list. I've even try to delete the 2 files after that but it still the same.

\* cries in italian *

WE'RE IN THE SAME F***ING REGION, WHY IS IT WORKING FOR ME BUT NOT YOU

\* smashes head against wall *

**Comment by i-am-bigmike** [comment]
I tried: closed Kodi, deleted the 2 files, rebooted... but nothing changed. I can see it downloads the suggested movies (the titles are "running" on the top/right of the screen), but the other lists are empty. In the best case I see the message "Loading next page", but then the list is still empty.

**Comment by Varstahl** [comment]
Seriously, do the following:

1. close Kodi
2. delete the content of `%AppData%\Kodi\userdata\addon_data\plugin.video.amazon-test` or `./.kodi/addon_data/plugin.video.amazon-test`
3. start Kodi
4. before entering the addon, right-click and open options
5. login
6. do **nothing** until it says "Welcome, <name>"
7. enter the addon
8. browse to any section
9. everything should work correctly. if it doesn't just lie to me ç_ç

jk on the lie part, but I believe something went horribly wrong somewhere the past couple days (and I have a couple ideas why) and everything broke.

Also, bonus points for anyone who uses PrimeVideo, opens `userdata\addon_data\plugin.video.amazon-test\config\accounts.lst`. searches for `"pv":` and tells me if the value next to PV is true or false.

**Comment by Allev8** [comment]
I don't know maaaaan
When I disinstall the amazon vod add-on i've to looking for something to delete manually? No, right? I've check if the folder is present after the disinstallation but there isn't anymore.

**Comment by Varstahl** [comment]
> When I disinstall the amazon vod add-on i've to looking for something to delete manually? No, right?

I'm not sure if the userdata persists, but in any case follow my [previous comment](https://github.com/Sandmann79/xbmc/issues/180#issuecomment-427519821) and tell me the result, whenever you can.

**Comment by i-am-bigmike** [comment]
Before deleting anything from "./.kodi/addon_data/plugin.video.amazon-test", "pv" is true to me.

**Comment by Varstahl** [comment]
> Before deleting anything from "./.kodi/addon_data/plugin.video.amazon-test", "pv" is true to me.

There goes nuked another possible explanation of the problem :| But thanks :)

**Comment by Allev8** [comment]
> 
> 
> > When I disinstall the amazon vod add-on i've to looking for something to delete manually? No, right?
> 
> I'm not sure if the userdata persists, but in any case follow my [previous comment](https://github.com/Sandmann79/xbmc/issues/180#issuecomment-427519821) and tell me the result, whenever you can.

I did and guess what? The same ;(
I got no welcome but only the standard pop up "logged in as ..." 
Is that what you mean right?

**Comment by Varstahl** [comment]
Yeah, that's it. I'm totally, completely and utterly at a fucking loss. We're in the same region, both using PrimeVideo, both from Italy, both resetting accounts and cache, my end works properly and yours doesn't. This doesn't make any fucking sense whatsoever. I've tested both Windows and Linux, and I've only had temporary problems 1 time in 10, all 10 of them I never experienced the issue you guys describe.

I don't know what to say/do. This doesn't make any sense to me.

**Comment by Varstahl** [comment]
The 0.5.9 provides a single fix for a network connection problem which at the moment doesn't change a single thing. From the accounts.lst I mentioned before, could you guys extract this values for me?

```
"save": "…",
"pv": …,
"name": "…",
"atvurl": "…",
"mid": "…",
"baseurl": "…"
```

> BTW, if it could be useful, I'm Italian too, and I'm using Raspberry.

Always said italians were the bane of my existance! :D ARGH! :D

**Comment by Allev8** [comment]
"save": "false", "pv": true, "name": "Alessandro", "atvurl": "https://atv-ps-eu.primevideo.com", "mid": "A3K6Y4MI8GDYMT", "password": "", "email": "", "baseurl": "https://www.primevideo.com"}]

This is mine

**Comment by i-am-bigmike** [comment]
"save": "false"
"pv": true
"name": "xxxxxx" (correct)
"atvurl": "https://atv-ps-eu.primevideo.com"
"mid": "A3K6Y4MI8GDYMT"
"baseurl": "https://www.primevideo.com"

**Comment by Varstahl** [comment]
Recitava il veccho adagio, in tema di muratori: “Il filo va a filo, il piombo va a piombo, e il muro va storto”.

Everything looks correct, nothing works. I have no idea how to solve this problem. It can't be login issues, because everything is correct. It can't be region, because I'm in Italy as well. It (shouldn't) be platform, because it works on linux and it shouldn't matter anyway. What's left?

The only thing left would be some weird addon conflict of some sorts. **Or** it could be a bad Kodi build, I'm still using the Alpha3 from last June on win7 and an even older one on linux…

One thing's for certain, I'm going to sleep. Maybe sleeping it over will help.

**Comment by i-am-bigmike** [comment]
I vecchi la sanno lunga! ;-)

My Kodi version was "old" (August), and I updated it only after the problem trying to resolve.

I don't think it's a login problem too, because "my list" and "my suggestions" are downloaded and I can watch the movies.

Dicono che la notte porti consiglio... buonanotte a tutti!

**Comment by Varstahl** [comment]
Updated Kodi to the latest version, still no issue. I wanted to try using LibreElec on the OrangePi I have laying around, but there's no love for H5, so that plan folded as well.

The only thing that comes to mind is maybe some vpn/caching compatibility issue of sorts… check if you have some weird or sketchy addons installed, and try to disable them. I'm quite at a loss, but I don't have a rPI, so I can't test it myself.

I've double checked everything, from source code static analysis to mere testing on both win7 and linux, everything is working correctly, I can't pinpoint anything wrong anywhere.

@i-am-bigmike you didn't post your logs, could you post them somewhere so that I may take a look?

**Comment by Allev8** [comment]
Ok, I just did a fresh new installation of libreelec on my pc, the only addon that i installed is amazon vod 0.5.9. Empty list again for each category. Could be an issue related to the ISP?

**Comment by Varstahl** [comment]
It makes no sense, though. But I **have** noticed that the DNS results vary greatly. What DNS servers are you guys using? Telecom? Fastweb? Google? OpenDNS? Although it *shouldn't* make a difference, unless they migrated the servers and the DNS caching is falling behind. No idea.

**Comment by Allev8** [comment]
Telecom.

**Comment by Varstahl** [comment]
> Ok, I just did a fresh new installation of libreelec on my pc

“!! You is of genius, just like Polan!!” I can use the LibreELEC ova to test it on PC, why didn't I think of it before.

> Telecom

So I assume you're using DNS like 151.99.0.100. Try changing DNS with OpenDNS ones, like 208.67.222.222/208.67.220.220, see if that changes anything. About **every** DNS I tested returns `52.85.24.54` as result for www.primevideo.com, except TIM's ones, returning `143.204.211.139` instead.

In the meanwhile I'm installing LibreELEC, maybe it's something to do with the platform.

**Comment by i-am-bigmike** [comment]
I did a new test. Removed the plugin.video.amazon-test directory and copied the sources from the commit of 26 September. Removed the 2 pvcp files in user data, leaving the other files untouched. Everything started to work again. :-)

If I understand correctly (I just took a quick look), in some of the last commits, there are changes related to the lists.

I noticed (with the not working version) most of the time I see the message "loading next page" when I try to open a list  (file primevideo.py). Maybe the problem is here somewhere. If you want to add some debug outputs I could try again and report the results.

Having installed a previous version, I don't have a log file now. If you need it I'll install the last version again.

As DNS (but I don't think it's the problem) I use Google.

**Comment by Allev8** [comment]
Ok, it seems not a DNS issue if you use google dns and i use telecom dns. 
Waiting for Varshtal and his libreelec now...

**Comment by Varstahl** [comment]
> I did a new test. Removed the plugin.video.amazon-test directory and copied the sources from the commit of 26 September

O_O

**NOTHING** without the updates prior to 29 of September should work. Around September 23-26th Amazon changed their frontend code. **Nothing** should work. If they work, you are somehow using and old version of PrimeVideo.com. That shouldn't even be possible. How could that be possible… What the heck is going on…

**Comment by i-am-bigmike** [comment]
@Allev8 if you know how to do it (I understand it's not a job for everyone), you could try an older version replacing the plugin.video.amazon-test files. It worked for me.

**Comment by Varstahl** [comment]
To (partially) revert it should be enough to just take the latest version and just replacing [primevideo.py](https://github.com/Sandmann79/xbmc/blob/70c366d7a0e72b2fd21abfea6d9c21a2ea191cbf/plugin.video.amazon-test/resources/lib/primevideo.py), you find the correct version linked.

I still want to know how the f*ck you three are travelling back in time though.

**Comment by Allev8** [comment]
i tried to replace primevideo.py but i can't even go inside the add-on now, i get error right away.
@i-am-bigmike No idea how to do unless you try to explain where i can find the older version and what folder should i have to replace exactly

**Comment by Varstahl** [comment]
Confirmed working perfectly with LibreELEC, I have to close this issue as invalid. I'll maybe edit the first post to notify users of the issue.

@Allev8 you can uninstall the addon, download the [0.5.7 zip from here](https://github.com/Sandmann79/xbmc/blob/70c366d7a0e72b2fd21abfea6d9c21a2ea191cbf/packages/plugin.video.amazon-test/plugin.video.amazon-test-0.5.7.zip), and install the addon manually by uploading it to (or downloading it from) LibreELEC.

**Comment by Allev8** [comment]
It says installation failed. I use kodi beta 3.

**Comment by Varstahl** [comment]
![LibreELEC](https://user-images.githubusercontent.com/284077/46575022-9fa95500-c9ad-11e8-995f-4ea1995b7b26.jpg)

For the “screen or didn't happen”, proof that the addon is working perfectly fine with LibreELEC. It's not a LibreELEC problem, it's not a Win7 problem, it's not a Linux problem. It (might) be an Italian problem (maybe?) or it could just be a server migration with only a portion of users moved. In any case, there's not much I can do, I can't support both the old and the new server code.

Further questions, out of curiosity:
* If you go to movies on primevideo.com, what URL does it send you to?
* If you go to primevideo.com's home, and search `-home"` in the sources, what do you find?

> It says installation failed. I use kodi beta 3.

Weird. Did you uninstall first?

**Comment by Varstahl** [comment]
@Allev8 from your log:

```
DEBUG: CAddonInstaller: installing from zip '/storage/emulated/0/Download/plugin.video.amazon-test-0.5.7.zip'
DEBUG: ZipManager: not a zip file!
```

You might have made a mistake downloading the file.

**Comment by Allev8** [comment]
Right click on your link and "salva destinazione con nome"
I tried even to open the link and click download but it's still the same...
Let me try on coreelec now...

**Comment by Allev8** [comment]
hold on, i can't even open the zip files in my pc. It says "archivio danneggiato o file sconosciuto"

**Comment by Varstahl** [comment]
My link brought you to the page where you had to download the zip, but you're right, I'll edit the comment with a direct link.

**Comment by Allev8** [comment]
Ok, finally it works again :D

**Comment by Varstahl** [comment]
Another proof that Italy's always living in the past :|

Seriously though, if you could answer the two questions under the screenshot, that'd be great.

**Comment by Allev8** [comment]
> Further questions, out of curiosity:
> 
>     * If you go to movies on primevideo.com, what URL does it send you to?

https://www.primevideo.com/storefront/movie/ref=atv_nb_sf_mv

>     * If you go to primevideo.com's home, and search `-home"` in the sources, what do you find?

I'm not sure how i can do it....

**Comment by Varstahl** [comment]
Ok, nevermind, I have no idea how the hell this is working for you guys, but I'm glad it does.

**Comment by Allev8** [comment]
We are in Italy man, everything is possible even if illogical, you know :)

**Comment by i-am-bigmike** [comment]
Quick update: installed the last version again... not working. Changed the primevideo.py with the version you provided, it works! (Of course I deleted the pvcp files evry time and recompiled primevideo.py after the replacement).
If you say nothing older than 29 September should work (and I don't have any doubt you could be wrong) I have no idea why I'm still in the past!

The movie link, but using my windows machine, so I can't say it's the same on Raspberry is : 
movie : https://www.primevideo.com/storefront/movie/ref=atv_nb_sf_mv
home : https://www.primevideo.com/storefront/home/ref=atv_nb_sf_hm

The only -home" I see in the source is : 
<div id="content" class="a-container  dv-storefront-page dv-page-home">

I don't think is what you was searching for.

Until it works with an old primevideo.py version, for me it's ok, and you can close the issue.

Thanks for your help and effort.

Ciao a tutti!!

**Comment by ameetmd** [comment]
Post upgrade to 0.5.8 on RPi3B+ and after deleting PVCatalog + PVVideoData, the movie list is working fine however the tv show list is still giving Amazon VOD error as seen below -

14:10:59.837 T:1522529008   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.KeyError'>
                                            Error Contents: u'parent'
                                            Traceback (most recent call last):
                                              File "/home/osmc/.kodi/addons/plugin.video.amazon-test/default.py", line 5, in <module>
                                                EntryPoint()
                                              File "/home/osmc/.kodi/addons/plugin.video.amazon-test/resources/lib/startup.py", line 83, in EntryPoint
                                                g.pv.Browse(None if 'path' not in args else args['path'])
                                              File "/home/osmc/.kodi/addons/plugin.video.amazon-test/resources/lib/primevideo.py", line 185, in Browse
                                                self._LazyLoad(node, nodeName)
                                              File "/home/osmc/.kodi/addons/plugin.video.amazon-test/resources/lib/primevideo.py", line 676, in _LazyLoad
                                                o[self._videodata[refUrn]['parent']] = self._videodata[self._videodata[refUrn]['parent']]
                                            KeyError: u'parent'
                                            -->End of Python script error report<--
14:11:00.081 T:1522529008 WARNING: CPythonInvoker(5, /home/osmc/.kodi/addons/plugin.video.amazon-test/default.py): the python script "/home/osmc/.kodi/addons/plugin.video.amazon-test/default.py" has left several classes in memory that we couldn't clean up. The classes include: N9XBMCAddon9xbmcaddon5AddonE,N9XBMCAddon7xbmcgui6DialogE

This is unusal & use based out on India.

**Comment by i-am-bigmike** [comment]
> I guess that when it will stop working you could upgrade…

Yes, I thought so too. ;-)

**Comment by CGarces** [comment]
I'm testing last addon affer remove PVCatalog and video

The navigation was
Your WatchList->TVShows.

[Full log](http://ix.io/1oyv)

**Comment by Varstahl** [comment]
> This is unusal & use based out on India

This could either mean there is a problem with the metadata or the code. Do me a favour, before the error there should be a request to `<domain>/catalog/GetPlaybackResources?<query>`: can you open it in a web browser and copy the content in a separate issue?

> Your WatchList->TVShows

Looks like an error in Amazon's servers. Could you provide me in a separate issue the HTML code of the page related to the Big Bang Theory season 10?

**Comment by JohnnyPicnic** [comment]
This is the error i'm getting with 5.9
`20:17:22.614 T:716 WARNING: [Amazon VOD] Unable to decode date "October 23, 2015": language "None" not supported
20:17:22.621 T:716  NOTICE: [Amazon VOD] postURL: https://atv-ps.primevideo.com/cdp/catalog/GetPlaybackResources?asin=amzn1.dv.gti.eeaaa367-aa4d-2c97-078d-8a6e33f00e94&deviceTypeID=AOAGZA014O5RE&firmware=1&deviceID=1c9682758ffcadcbb1467dc0a884eb01201411ab53dc61a1c557a66b&marketplaceID=ART4WZ8MWBX2Y&format=json&version=1&gascEnabled=true&resourceUsage=ImmediateConsumption&consumptionType=Streaming&deviceDrmOverride=CENC&deviceStreamingTechnologyOverride=DASH&deviceProtocolOverride=Https&deviceBitrateAdaptationsOverride=CVBR%2CCBR&audioTrackId=all&videoMaterialType=Feature&desiredResources=CatalogMetadata&titleDecorationScheme=primary-content
20:17:23.157 T:716   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.AttributeError'>
                                            Error Contents: 'NoneType' object has no attribute 'group'
                                            Traceback (most recent call last):
                                              File "C:\Users\User\AppData\Roaming\Kodi\addons\plugin.video.amazon-test\default.py", line 5, in <module>
                                                EntryPoint()
                                              File "C:\Users\User\AppData\Roaming\Kodi\addons\plugin.video.amazon-test\resources\lib\startup.py", line 83, in EntryPoint
                                                g.pv.Browse(None if 'path' not in args else args['path'])
                                              File "C:\Users\User\AppData\Roaming\Kodi\addons\plugin.video.amazon-test\resources\lib\primevideo.py", line 185, in Browse
                                                self._LazyLoad(node, nodeName)
                                              File "C:\Users\User\AppData\Roaming\Kodi\addons\plugin.video.amazon-test\resources\lib\primevideo.py", line 640, in _LazyLoad
                                                self._videodata[refUrn]['title'] = Unescape(title.group(1))
                                            AttributeError: 'NoneType' object has no attribute 'group'
                                            -->End of Python script error report<--
20:17:23.249 T:716 WARNING: CPythonInvoker(8, C:\Users\User\AppData\Roaming\Kodi\addons\plugin.video.amazon-test\default.py): the python script "C:\Users\User\AppData\Roaming\Kodi\addons\plugin.video.amazon-test\default.py" has left several classes in memory that we couldn't clean up. The classes include: class XBMCAddon::xbmcgui::Dialog,class XBMCAddon::xbmcaddon::Addon
20:17:23.254 T:2068   ERROR: XFILE::CDirectory::GetDirectory - Error getting plugin://plugin.video.amazon-test/?mode=PrimeVideo_Browse&path=root-%2f%2f-TV%20Shows
20:17:23.255 T:3140   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.amazon-test/?mode=PrimeVideo_Browse&path=root-%2f%2f-TV%20Shows) failed`

**Comment by pletexier** [comment]
Hello !
Same problem in France. With 5.9, all categories are empty.
With 5.7, the plugin is awfully long to load.

Regards,

Patrice

**Comment by Varstahl** [comment]
Stick to 5.7 until it works, then switch to 5.9. There's no workaround for that.

**Comment by i-am-bigmike** [comment]
Today I spent some time to debug the 5.9.

For "Movie and series list" (as example) it searches the string "dvui-beardContainer", while the old version has as search string "av-result-cards" for the same category.

Using my windows machine and firefox as browser, I find "av-result-cards" in the page source and not the new one.
The same for "av-episode-list" and other search strings.

This explains why the lists are empty.

But why "dvui-beardContainer" works for someone and not for others? Is Amazon migrating ALL (?) users to new servers? Or there is something else?

**Comment by Varstahl** [comment]
The problem goes deeper than that, because after that you need to look for different triggers for a lot of stuff. Different triggers for movie data, different triggers for episode data, different triggers for containers. And in all of this I don't even know if it's possible to support both at the same time, I wasn't expecting for it to be a partial migration, so I just don't know.

>  Is Amazon migrating ALL (?) users to new servers? Or there is something else?

Beats me. A few users reported the content broken, mostly, but not only, across Europe. I, too, was affected, so I believed it was a full scale UI change, so I took my time and fixed it for the new layout. This was in Friday, where everyone's addon completely broke. Then, on Monday evening, a new change appeared and completely broken my addon while I was working on it, and that's where the userbase split, but I didn't know at the time. I knew everything got screwed up, so I trashed my previous work, delete a lot of code that used Amazon-JSON-Streaming objects, and replaced it with HTML parsing. The problem is, that some have the old version still, some have the new one, and the reason why is still unclear.

But you DO have the new version. When you tested through your browser, I'm fairly sure you and @Allev8 found content with the new UI. Which, unless completely mistaken, means that on the browser you have the updated UI, while on Kodi you got the old one.

I'm unable to test and unable to fetch content with the old one. I thought both of supporting the old layout, and of taking the latest changes and backporting it to 0.5.7, but I don't even know if any of that would work.

**Comment by i-am-bigmike** [comment]
Thanks for your reply.

> The problem goes deeper than that, because after that you need to look for different triggers for a lot of stuff. Different triggers for movie data, different triggers for episode data, different triggers for containers. 

I saw there are a lot of triggers, I just tried to keep the message shorter :-)

> The problem is, that some have the old version still, some have the new one, and the reason why is still unclear.
> I'm unable to test and unable to fetch content with the old one. I thought both of supporting the old layout, and of taking the latest changes and backporting it to 0.5.7, but I don't even know if any of that would work.

I understand this causes a lot of confusion and it's not easy (if not possible) the keep the program updated and fully working.

You told me, and confirmed, my browser (on the windows machine) sees the new UI. How can I check, on various computers, if it's the new or the old?
And, (I write this just to try to find an idea or a solution), if my browser is connected to the new UI, why (I don't expect you have the answer) in the page's sources I find many of the old tags, like as example "av-result-cards" and not the new "dvui-beardContainer" ? (I tried with other tags too, but I don't remember the exact names).

Let's hope it's just a migration to new servers, and it will be completed soon.

Just an idea...
thinking to the app for Android... this must work both with newer and older servers. I think Amazon has the way to get the movie's list "easily", just like sending a request and receiving a json encoded reply...
Maybe it could be possible to find this protocol... ;-)

**Comment by Varstahl** [comment]
> You told me, and confirmed, my browser (on the windows machine) sees the new UI. How can I check, on various computers, if it's the new or the old?

The old versions didn't have `/storefront/` URLs, the new one does. The old version used `av-identifier`, the new one `dv-identifier`. Of course I'm going off the top of my head. The old version also had `span.title` or something similar, the new one doesn't. I guess that somehow the current version is a hybrid, or there's something more to it. Version 0.5.7 still uses JSON-AmazonUI-Streaming object for its main categories, so possibly that's why it's still working somehow. The main difference however should be in categories list, for example [this page](https://www.primevideo.com/search/ref=atv_hm_hom_c_fsjguc_3_seemre?query=p_n_entity_type%3DTV%20Show%26search-alias%3Dinstant-video%26index%3Deu-amazon-video-euro%26adult-product%3D0%26bq%3D(and%20sort:'featured-rank'%20(and%20studio:'amazon%20studios'%20(and%20(not%20(or%20av_primary_genre:'children'%20av_primary_genre:'family'))%20(not%20(or%20av_secondary_genre:'children'%20av_secondary_genre:'family')))))%26av_offered_in_territory%3DIT%26qs-av_request_type%3D4%26qs-is-prime-customer%3D2&ie=UTF8):

![TV Series](https://user-images.githubusercontent.com/284077/46701236-54967880-cc1f-11e8-98a9-ce27e2dd9d06.png)

The old layout was list based, the new one is grid based. If you have the grid, it's the newest layout and 0.5.7 should crash and burn. I think this is the best method, but of course I don't know if that's actually true.

> thinking to the app for Android... this must work both with newer and older servers. I think Amazon has the way to get the movie's list "easily", just like sending a request and receiving a json encoded reply...
> Maybe it could be possible to find this protocol... ;-)

I wanted to reverse engineer the protocol for smart TVs, since I have one available, but as far as I could tell it's using pure pre-cooked server rendered HTML5. I know this kind of for a fact, because out of the many test and usage I had with the JSON-AmazonUI-Streaming objects, at times (and at random) the JSON variant would screw up, fail to retrieve the localized version, and return the english version instead. That meant that while browsing with their app, or a browser, HTML5 version was always perfect, when using JSON APIs (I guess now completely deprecated) it often failed.

Also, I just came to think of it, you're looking at this in the wrong way: they don't need to understand what they're shown, they just react. Whether it's the old UI or the new UI, the JavaScript management goes along with the UI they're using. It's just *our* problem, since we have to extrapolate the data.

**Comment by i-am-bigmike** [comment]
Thanks for your explanations!
Yes, I see the "grid" version, while Kodi works with the old version... Crazy! :-)

As you pointed, I was wrong, I didn't look at the problem from the right side. The app doesn't need to interpreter anything... just show what the server sends. :-(

**Comment by mymo1982** [comment]
Hello, I've been going through a similar issue as described in this topic.
Almost of the lists work, but not all: I've been trying to watch The Romanoffs, but only the trailer is showed in the list of episodes, whereas there are 2 already available through the Amazon app and website.

This is happening on my Android box, I haven't tested on Windows yet. I'm in Italy.

**Comment by Varstahl** [comment]
@mymo1982 completely unrelated issue. What you're experiencing is aggressive caching affecting the results. I've been changing the backend severely and I didn't implement a way to re-download portions of the data and metadata of movie and series, including the episodes list. For the time being you need to delete the *.pvdp file, which will reset all video data downloaded. Inconvenient, but that's it.

**Comment by mymo1982** [comment]
It's inconvenient but at least it worked, thanks.

**Comment by Varstahl** [comment]
Yeah, I've been changing a lot in the back end recently, now this sort of cache update could be made to work, but requires some more development which I didn't put in yet, but it's on its way.

**Comment by mymo1982** [comment]
Now the tv series list doesn't load. The film list, kid list, and search work, I can even search series and watch episodes with no issues, but when trying to load the tv series list all I get is an Error (and the suggestion to read the log file)

**Comment by mo123** [comment]
Also just getting an error now if loading the tv shows if country is set to Auto.
When set to UK, US or Far East, I can see all the tv shows but it says, not available in your country or you need to buy the content.

**Comment by Varstahl** [comment]
@mo123 @mymo1982 @i-am-bigmike @Allev8 and anyone who can help.

I need a favour if I am to get to the bottom of this. I need the following HTML file contents from these pages:
* [PrimeVideo homepage](https://www.primevideo.com/)
* [Series page](https://www.primevideo.com/storefront/tv/)
* [Expanded category page](https://www.primevideo.com/search/ref=atv_tv_hom_c_fsjguc_3_seemre?query=p_n_entity_type%3DTV%20Show%26search-alias%3Dinstant-video%26index%3Deu-amazon-video-euro%26adult-product%3D0%26bq%3D(and%20sort:'featured-rank'%20(and%20studio:'amazon%20studios'%20(and%20(not%20(or%20av_primary_genre:'children'%20av_primary_genre:'family'))%20(not%20(or%20av_secondary_genre:'children'%20av_secondary_genre:'family')))))%26av_offered_in_territory%3DIT%26qs-av_request_type%3D4%26qs-is-prime-customer%3D2&ie=UTF8)
* Any season page, [like this](https://www.primevideo.com/detail/0T1R1MXA75ZCR0C3247CHPIS3X/) for example, but from your own listings.

There are many differences across the listings that I can't get to the bottom of if I don't have your versions of the pages.

**Comment by mymo1982** [comment]
Do we need to save these pages through our browser? If so, do we need to be logged in?
Thanks for your support

**Comment by Varstahl** [comment]
Yes to both the answers :)

**Comment by mymo1982** [comment]
There you go :)

https://drive.google.com/open?id=1EWTj_LofmI04M6Z0RL5PsNJY1f9kuzID

**Comment by ameetmd** [comment]
Hope this helps -
https://1drv.ms/f/s!Al9Ia2-XiIirhLsBA1CSGIWoaN0paA

    On ‎Wednesday‎, ‎October‎ ‎24‎, ‎2018‎ ‎04‎:‎58‎:‎09‎ ‎PM‎ ‎IST, mymo1982 <notifications@github.com> wrote:  
 
 
There you go :)

https://drive.google.com/open?id=1EWTj_LofmI04M6Z0RL5PsNJY1f9kuzID

**Comment by Allev8** [comment]
http://www.easybytez.com/4staekacd3zy

I hope is what are you looking for..

**Comment by mo123** [comment]
Hope it helps.
https://mega.nz/#!5IdGiKgL!IqZ_05QuSesPMsg9WYRc2epKnk7nMGiQZOP_amH7p_Y

**Comment by i-am-bigmike** [comment]
I arrived too late to post my html contents, but in time to try the new version. :-)
Although unnecessary, I deleted the 2 cache files, downloaded the new primevideo.py and...

IT WORKS!!

I tried with some series and some movie categories, everything worked fine without any error. Thansk a lot **Varstahl** for your support!

**Comment by mymo1982** [comment]
I confirm that it works! Thanks!

Do you think the lists will be automatically updated with new episodes as they come along?

**Comment by Varstahl** [comment]
For incomplete seasons, no, you need to wait the implementation of the refresh or manually delete the cache files to update (eg. The Romanoffs). For new series yes, it will update by itself.

**Comment by ezu64** [comment]
Thank you Varstahl . It works fine ! I've an unrelated issue. Some series like "The Grand Tour" have no sound track . It's e very weird thing!

**Comment by Varstahl** [comment]
@ezu64 same issue as #164, causes still unknown.

**Comment by rytecbe** [comment]
Good afternoon

I have followed some tips and tryouts from above.
previous I had Amazon VOD 0.7.0ter installed on my raspberry pi with osmc.
But also I have the problem that I cannot see the tvseries episodes.

Then I tried using 0.5.7 but then I got the message "unable to find the main primevideo.com navigation section" 

I tried clearing the catalog file and also used that newer primevideo.py file which you had posted but then again no episodes to select for starting.

What should I do and post here for you so you could have a look at it?

Thanks for your help.

**Comment by Varstahl** [comment]
@rytecbe 0.6.7 won't work, the issue is discussed in #285 but I haven't been able to work on it.

**Comment by rytecbe** [comment]
> 
> 
> @rytecbe 0.6.7 won't work, the issue is discussed in #285 but I haven't been able to work on it.

@Varstahl Thanks for the quick reply, I appreciate that. Should I update Amazon VOD then to 0.7.0 or will that be worse?

BR
Ryan

**Comment by Varstahl** [comment]
0.7.0ter is the most up to date. For some users shows are a problem, but I haven't had time to investigate the issue yet. As soon as it's fixed it should automatically update.

**Comment by Surajpd** [comment]
Hi Varstahl
With 7.4 Amazon in OSMC on Raspberry pi 3, I still find the movie, tv lists blank. I scanned through the thread and couldnt find a solution yet. I am in India. Could you direct me to the solution if this is closed. Thanks in advance.
https://github.com/Sandmann79/xbmc/issues/180#issuecomment-432981678

**Comment by Varstahl** [comment]
You can find a temporary solution in #322.

**Comment by Surajpd** [comment]
Thanks a ton for the prompt response. Will try this out
